src/Orleans.Runtime/Catalog/ActivationData.cs

```diff
@@ -1270,7 +1270,7 @@ private async ValueTask<bool> RegisterActivationInGrainDirectoryAndValidate()
                 try
                 {
                     var result = await _shared.InternalRuntime.GrainLocator.Register(Address);
-                    if (Address.Equals(result))
+                    if (Address.Matches(result))
```

Comment:
cc @benjaminpetit PTAL.
The `CachedGrainLocator` sets the `GrainAddress.MembershipVersion` property on the way through, so the result will never match (since the `MembershipVersion` on `Address` here is the default value). I thought about making `Address` mutable, but I think this is a safer change. `MembershipVersion` should only be used for silo liveness tests, anyhow (AFAICR).

See https://github.com/dotnet/orleans/blob/d423f2a1088a4c0d1333c9e6893b7cdf625d6da1/src/Orleans.Runtime/GrainDirectory/CachedGrainLocator.cs#L92

Reply:
Yes, that's definitely how it should have been done